tools/autograd/gen_autograd_functions.py

```diff
@@ -993,14 +1032,38 @@ def emit_derivative(
         f"{T} {x}"
         for T, x in zip(apply_functional_args_ref_types, apply_functional_args)
     ]
+    get_packed_args = "\n".join(
+        f"packed_args.pack({name});" for name in apply_functional_args
+    )
+    unpack_ivalues = []
+    for typ, name in zip(apply_functional_args_ref_types, apply_functional_args):
+        if typ.endswith("&"):
+            typ = typ[:-1]
+        unpack_ivalues.append(f"auto {name} = packed_args.unpack<{typ}>();")
+
+    schema_args = [f"std::array<bool, {len(input_name_to_idx)}>"]
+    for typ in apply_functional_args_ref_types:
+        if typ.endswith("&"):
```

Comment:
In theory, the torchgen infra should be able to help you avoid string parsing as much as possible.

In this case, it looks like we generated `apply_functional_args_ref_types` from a bunch of `CTypes` ([link](https://github.com/pytorch/pytorch/blob/main/torchgen/api/types/types_base.py#L75)). You should be able to do something like:
```
for type in types:
    schema_args.append(type.remove_const_ref().ctype())
```

If you think your existing string parsing got you passed all of the edge cases / the existing string parsing makes this a painful refactor then np.

Reply:
It was not clear to me that torchgen has a SavedVariable type (that looked special cased in the autograd codegen)